lib/ios-fs-helpers.js

```diff
@@ -97,13 +97,16 @@ export async function pullFolder(afcService, remoteRootPath) {
       readStream.pipe(writeStream);
       if (pullPromises.length >= MAX_IO_CHUNK_SIZE) {
         await B.any(pullPromises);
+        for (let i = pullPromises.length - 1; i >= 0; i--) {
+          if (pullPromises[i].isFulfilled()) {
+            pullPromises.splice(i, 1);
+          }
+        }
       }
-      // @ts-ignore
-      _.remove(pullPromises, (p) => p.isFulfilled());
     });
     // Wait for the rest of files to be pulled
     if (!_.isEmpty(pullPromises)) {
-      await B.all(pullPromises);
+      await B.all(pullPromises.filter((p) => !p.isFulfilled()));
```

Comment:
it's fine to keep it as it was. Already resolved promises are not going to be awaited anyway